core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -428,21 +494,71 @@ private[sql] final case class ColumnTableScan(
           // initialize the decoder and store the starting cursor position
           $cursor = $decoder.initialize($buffer, $planSchema.apply($index));
         """)
-      bufferInitCode.append(
-        s"""
+
+
+      if (isWideSchema) {
+        if (bufferInitCode.length > 1024) {
+          bufferInitCodeBlocks.append(bufferInitCode.toString())
+          bufferInitCode.clear()
+        }
+
+        bufferInitCode.append(
+          s"""
+          final $decoderClass $decoderVar = $decoder;
+          $bufferVar = ($buffer == null || $buffer.isDirect()) ? null
```

Comment:
This code can be refactored to declare bufferVar of non-wide case before the code block. Rest of the part should remain same.